### PR TITLE
Supporting Multi-class, ignore index, and different input shapes

### PR DESCRIPTION
**Pull request by msalhab96**
This pull request solves the issues below:
- Multi-class Focal Loss issue #7 
- adding ignore index parameter #6 
- ability to support different input shape

To make sure the code is running as expected I did the below:
trained a model using CrossEntropyLoss from PyTorch, then used the Focal Loss implemented in the repo and set gamma to 1 and the result matched.

I guess the next step after this pull request is to add unit tests in order to make the code maintainable.

@mathiaszinnen Thanks for the project, and looking forward to hearing your feedback.

### CODE REVIEW

**Comment by mathiaszinnen** [comment]
Thanks a lot for taking care of this!
